### DISCUSSION TITLE
Black ImGui Demo screen and Error Message in MSVC Command Window

### DISCUSSION THREAD

**Discussion by AugustusTurcarum** [ocornut/imgui#6803] [New Users Build/Link/Run issues ONLY!]
Hi everyone,
I am using latest ImGui Docking Branch , GLFW 3.3.8 and GLAD libraries
My System Specifications:
Windows 11 
AMD Ryzen 7 6800H with Radeon Graphics
Visual Studio 2022

I am following TheCherno Game Engine Series from his Youtube channel. He is using ImGui and GLFW for the Game Engine HAZEL that he is creating.
I am following the same steps as he progress.However I am experiencing an issue since the Game Engine starts using Docking Branch of ImGui library while he is still running without any issue.
I am getting an Error Display in MSVC Debug command window when I call "ImGui_ImplOpenGL3_NewFrame()" function inside the code below under ImGuiLayer::Begin function !
When I follow the Debug up to the Error Display, it is going through  imgui_impl_opengl3.cpp  --> ImGui_ImplOpenGL3_CreateDeviceObjects() --> CheckShader() function
Due to this Error message, ImGui::ShowDemoWindow function makes a black Imgui window appear

The link for Cherno's ImGuiLayer.cpp code : [https://github.com/TheCherno/Hazel/blob/master/Hazel/src/Hazel/ImGui/ImGuiLayer.cpp](url)

I am still following previous version of his code. It means my ImGuiLayer.cpp file and his is not exactly the same, but the way we call ImGuiLayer::Begin() function is the same.
My following of Cherno's code, a part of  ImGuiLayer.cpp is below:

//////////////////////////////////////////////////////////////////////////////////
`#include "hzpch.h"
#include "Hazel/ImGui/ImGuiLayer.h"

#include <imgui.h>

//#define IMGUI_IMPL_API
#include <backends/imgui_impl_glfw.h>
#include <backends/imgui_impl_opengl3.h>

//#include "Platform/OpenGL/ImGuiOpenGLRenderer.h"
#include "Hazel/Core/Application.h"

// THIS PART IS TEMPORARY
#include <GLFW/glfw3.h> 
#include <glad/glad.h>

namespace Hazel
{

	ImGuiLayer::ImGuiLayer()
		: Layer("ImGuiLayer")
	{
	}

	ImGuiLayer::~ImGuiLayer()
	{
	}

	void ImGuiLayer::OnAttach()
	{
		HZ_PROFILE_FUNCTION();

		// Setup Dear ImGui Context
		IMGUI_CHECKVERSION();
		ImGui::CreateContext();
		ImGuiIO& io = ImGui::GetIO(); (void)io;
		io.ConfigFlags |= ImGuiConfigFlags_NavEnableKeyboard;  // Enable Keyboard Controls
		// io.ConfigFlags |= ImGuiConfigFlags_NavEnableGamepad;  // Enable Gamepad Controls
		io.ConfigFlags |= ImGuiConfigFlags_DockingEnable;  // Enable Docking
		io.ConfigFlags |= ImGuiConfigFlags_ViewportsEnable;  // Enable Multi-ViewPort / Platform Windows
		//io.ConfigFlags |= ImGuiConfigFlags_ViewportsNoTaskBarIcons;
		//io.ConfigFlags |= ImGuiConfigFlags_ViewportsNoMerge;

		// Set Up ImGui Style
		ImGui::StyleColorsDark();
		//ImGui::StyleColorsClassic();
		
		// When viewports are enabled, we tweak WindowRounding/WindowBg so platform windows can look identical 
		ImGuiStyle& style = ImGui::GetStyle();
		if (io.ConfigFlags & ImGuiConfigFlags_ViewportsEnable)
		{
			style.WindowRounding = 0.0f;
			style.Colors[ImGuiCol_WindowBg].w = 1.0f;
		}

		Application& app = Application::Get();
		GLFWwindow* window = static_cast<GLFWwindow*>(app.GetWindow().GetNativeWindow());

		// Setup Platform/Renderer bindings
		ImGui_ImplGlfw_InitForOpenGL(window, true);
		ImGui_ImplOpenGL3_Init("#version_410");
	}

	void ImGuiLayer::OnDetach()
	{
		HZ_PROFILE_FUNCTION();

		ImGui_ImplOpenGL3_Shutdown();
		ImGui_ImplGlfw_Shutdown();
		ImGui::DestroyContext();
	}

	void ImGuiLayer::OnEvent(Event& e)
	{
		ImGuiIO& io = ImGui::GetIO();
		e.Handled |= e.IsInCategory(EventCategoryMouse) & io.WantCaptureMouse;
		e.Handled |= e.IsInCategory(EventCategoryKeyboard) & io.WantCaptureKeyboard;
	}

	void ImGuiLayer::Begin()
	{
		HZ_PROFILE_FUNCTION();

		ImGui_ImplOpenGL3_NewFrame();
		ImGui_ImplGlfw_NewFrame();
		ImGui::NewFrame();
	}

	void ImGuiLayer::End()
	{
		HZ_PROFILE_FUNCTION();

		ImGuiIO& io = ImGui::GetIO();
		Application& app = Application::Get();
		io.DisplaySize = ImVec2((float)app.GetWindow().GetWidth(), (float)app.GetWindow().GetHeight());

		//Rendering
		ImGui::Render();

		ImGui_ImplOpenGL3_RenderDrawData(ImGui::GetDrawData());

		if (io.ConfigFlags & ImGuiConfigFlags_ViewportsEnable)
		{
			GLFWwindow* backup_current_context = glfwGetCurrentContext();
			ImGui::UpdatePlatformWindows();
			ImGui::RenderPlatformWindowsDefault();
			glfwMakeContextCurrent(backup_current_context);
		}

	}

	//void ImGuiLayer::OnImGuiRender()
	//{

	//	static bool show = true;
	//	ImGui::ShowDemoWindow(&show);

	//	//ImGui::Begin("");
	//	//ImGui::End();

	//}	
}`
//////////////////////////////////////////////////////////////////////////////////

The Error message in MSVC Debug command window is:

![Error_Screen](https://github.com/ocornut/imgui/assets/144597108/7b5ce757-d943-4e97-bfd0-5d1d6be0f97b)

It is always activating the error message inside CheckShader() function. I don't know if the reason is AMD Graphic Card or not . I checked FAQ, previous discussions but couln't find the solution.
Can someone help me with the solution for this problem ?
Any help would be much appreciated

Sincerely Yours

**Comment by PathogenDavid** [comment]
The version directive you pass to `ImGui_ImplOpenGL3_Init` is directly prepended to the internal shaders. As the error states, `#version_410` is not a valid GLSL version directive.

Also in the future please ensure you use correct Markdown formatting in your questions, your code is very annoying to read as it is currently written.
